test/layout/slots.js

```diff
@@ -0,0 +1,152 @@
+const { config } = require('hardhat');
+const { expect } = require('chai');
+const fs = require('fs');
+const os = require('os');
+const path = require('path');
+
+const extractStorageLayout = require(path.join(config.paths.root, 'scripts/extract-storage-layout'));
+
+const cleanupType = type => {
+  return (
+    type
+      // t_mapping(t_uint256,t_struct(CoverInfo)32880_storage)
+      // ->
+      // t_mapping(t_uint256,t_struct(CoverInfo)_storage)
+      .replace(/\)\d+/, ')')
+      // 2. replace contract types with t_address
+      .replace(/t_contract\([^)]+\)/, 't_address')
+      // 2. ignore payable specifier for addresses
+      .replace(/t_address_payable/, 't_address')
+  );
+};
+
+describe('Storage layout', function () {
```

Comment:
have we verifed the quality of this test by intentionally breaking layout to see the differences?